### ISSUE TITLE
copySourcePathAssets not copying Assets.car along with some other files?

### ISSUE THREAD

**Issue by cgascons** [BowlerHatLLC/vscode-as3mxml#381]
Hi Josh! I just started to try to migrate all our projects to Visual Studio using your extension, first of all, thank you very much about it, it really does look great so far.

I've actually have quite a bunch of questions but I will try to figure them out all as I keep trying to integrate my Flash Builder projects onto this new system.

The one question I'd like to ask by now is regarding the copySourcePathAssets. I understand that, according to the [documentation](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#copysourcepathassets) that option: 

> Copies all files in the source path that don't have .as or .mxml extensions into the output directory.

Actually to create a debug build and run it on the simulator it does look like it's working fine, but whenever I try to package it to debug or release on a device, the .ipa size doesn't look good, after decompressing it I found that not all assets were added, I actually notice that when trying to build the release package, a warning came up stating that the Assets.car asset wasn't being packaged in the app, which made me try to add it directly into the "files" option, and that way it did not warn me anymore.

My current configuration for my test project is quite complicated to be honest (it uses 3 custom libraries, which have also been added as projects in Visual), but I think it should still work. This is the asconfig.json and the launch.json files:

**asconfig.json**:
```json
{
	"compilerOptions": {
		"output": "bin/MainGame.swf",
		"source-path": [
			"src",
			"media",
			"media_on_demand",
			"media_on_demand_project",
			"../TCGEngine/externals/gamesparks-as3-sdk/src"
		],
		"library-path": [
			"../TCGEngine/libs",
			"../TCGEngine/bin/TCGEngine.swc",
			"../FSMobileLibrary/bin/FSMobileLibrary.swc",
			"../FSAirLibrary/bin/FSAirLibrary.swc"
		],
		"external-library-path": [
			"../TCGEngine/anes/common"
		]
	},
	"files": [
		"src/MainGame.as"
	],
	"application": "src/MainGame-app.xml",
	"copySourcePathAssets": true,
	"additionalOptions": "-swf-version=42",
	"airOptions": {
		"extdir": [
			"../TCGEngine/anes/common"
		],
		"ios": {
			"listen": true,
			"output": "dist/WWIITCG.ipa",
			"signingOptions": {
				"storetype": "pkcs12",
				"keystore": "../TCGEngine/cert/ios_profile_dev.p12",
				"provisioning-profile": "cert/ww2_dev.mobileprovision"
			}
		},
		"files": [
			{
                "file": "src/icons",
				"path": "icons"
			},
			{
				"file": "src/Assets.car",
                "path": "Assets.car"
			}
		]
	},
	"config": "airmobile"
}
```

**launch.json**:

```json
{
   "version": "0.2.0",
    "configurations": [
        {
            "type": "swf",
            "request": "launch",
            "name": "Launch SWF",
            "program": "bin/MainGame-app.xml",
            "preLaunchTask": "ActionScript: compile debug - asconfig.json"
        },
        {
            "type": "swf",
            "request": "launch",
            "name": "Launch SWF (iPhone)",
            "profile": "mobileDevice",
            "screensize": "iPhoneRetina",
            "screenDPI": 326,
            "versionPlatform": "IOS",
            "preLaunchTask": "ActionScript: compile debug - asconfig.json"
        },
        {
            "type": "swf",
            "request": "attach",
            "name": "Install and Attach (iOS)",
            "platform": "ios",
            "connect": true
        }
    ]
}
```

I suppose I should not need to add the `media`, `media_on_demand`, and `media_on_demand_project` by hand into the "files" option, right?

Also, just as extra information, whenever I try to package and install the app on my device I get this message:

```
Preparing to install Adobe AIR application...
Installing Adobe AIR application...
Installing app on device failed for platform "ios" with status code: 12.
```

Which, according to Adobe is: 
```
Error: 12 | Invalid input | Verify file paths and other arguments passed to ADT on the command line.
```

Also, if I attempt to install the generated .ipa directly to the device via xCode the error generated is: 
```
A signed resource has been added, modified, or deleted
```

Thanks beforehand for your support!

**Comment by joshtynjala** [comment]
> I suppose I should not need to add the media, media_on_demand, and media_on_demand_project by hand into the "files" option, right?

Actually, yes. All assets that you want to include in your AIR app must be included in the `files` field of the `airOptions` section.

`copySourcePathAssets` only copies files to the same folder where your SWF is created. It does not affect how your AIR app is packaged. Technically, I guess it is most useful for Flash Player projects.

**Comment by joshtynjala** [comment]
> A signed resource has been added, modified, or deleted

I found this question with some potentially useful answers on Stack Overflow:

https://stackoverflow.com/questions/2157964/receive-message-a-signed-resource-has-been-added-modified-or-deleted-when-tr

Several people mentioned that they fixed this issue by removing special characters, like `?`, `(`, and `)` from the name of their application. You might want to check what's contained in your *src/MainGame-app.xml* file.

It might not necessarily be that specifically, but this seems to hint to me that there's something about your packaged app that iOS isn't happy with.

**Comment by cgascons** [comment]
> Actually, yes. All assets that you want to include in your AIR app must be included in the `files` field of the `airOptions` section.

I see, it looks like all the assets are packaged properly now, thanks for the info.

> Several people mentioned that they fixed this issue by removing special characters, like `?`, `+`, `(`, and `)` from the name of their application. You might want to check what's contained in your _src/MainGame-app.xml_ file.

Regarding this, I actually googled it before creating this issue but forgot to mention about it, sorry about that! I did review my app descriptor and could not find any special characters in it, weird. 

Anyways something has happened, after adding the media folders into the `fields` section, the error message changes, now it's returning the following error:

`Preparing to install Adobe AIR application...
Installing Adobe AIR application...
Installing app on device failed for platform "ios" with status code: 14.`

, which led me to the issue #221. 

I have followed all your suggestions in that post but I'm still getting the same error over and over (I'm even getting the blank error box). After trying again to install the .ipa via the xCode I'm getting the same `A signed resource has been added, modified, or deleted` error message though, perhaps it's a mobileProvision error? I've tried creating both a debug and a release versions but none of them work, I also recall being able to debug on iOS running Flash Builder on a test iPhone before.

Not sure how to proceed or what else to try.

**Comment by joshtynjala** [comment]
Yeah, I guess it could be an issue with the mobileprovision. If it is, installation should fail if you try to use it in another IDE too.

This is definitely something I've never encountered. I don't really have anything to suggest. It might be worth trying a simpler test project. If that works try your main project again, but start disabling things. ANEs, for instance, seem like they could cause this sort of issue.

**Comment by cgascons** [comment]
Thanks Josh, will let you know what was the cause as soon as I find it

**Comment by cgascons** [comment]
Hi Josh, I've kept trying without luck. I have been able to package a very tiny sample project which still doesn't allow me to debug an app on iOS, please find it attached to this post. 

The project is as simple as I could make it, notice that I have not added my development profile nor my provisioning profile, I suppose you can use yours for the test if you don't mind. 

It's just a "Hello world" app.

Thanks,
[Test.zip](https://github.com/BowlerHatLLC/vscode-as3mxml/files/3334824/Test.zip)

**Comment by joshtynjala** [comment]
I'll try to look at this soon!

**Comment by cgascons** [comment]
Thanks, looking forward hearing back from you

**Comment by joshtynjala** [comment]
@cgascons Your sample project works correctly on my computer. First, I ran the "Adobe AIR: package iOS debug - asconfig.json" task. Then, I launched the "Install & Attach Adobe AIR (iOS)" configuration to install and debug the app on my iOS device.

The only change that I made was to update the paths to my *.mobileprovision* and *.p12* files in *asconfig.json*.

**Comment by joshtynjala** [comment]
I confirmed myself that it seems to open the correct URL in Firefox and Chrome. It's either a Safari issue, or an issue with the debugger in the SDK. It's actually the debugger that launches the specified app and not VSCode or the extension.

**Comment by cgascons** [comment]
I don't understand then what could I be doing wrong, I have made sure both devices where I'm running these tests are added in the profile, re-downloaded the certificate and the mobileprovision from the Apple member center and I'm still getting the same issue. 

Could it be related to the current java version that I'm using? (1.8.0_112). I use that one because one of the certificates we use for an Android project was made 7 years ago and isn't 2048, so I have to use the last java version that allows using these old certificates and there's no way (that I know of) to "upgrade" that certificate to a newer version without having to create a new application on the store.

**Comment by joshtynjala** [comment]
I wouldn't expect which particular Java 1.8 update that you have installed to matter, but I guess that you could try a different one easily enough.

**Comment by cgascons** [comment]
Tried on 2 different windows computers (with different Java versions), also on my Mac again and nothing, I just can't get it to debug on an iPhone or iPad (I have also updated them to their latest available software version).

I even checked the entitlements that the app needs by using:
```codesign -d --entitlements :- "Payload/Main.app"```

And checked what entitlements are included in the provision with
```security cms -D -i "Payload/Main.app/embedded.mobileprovision"```

Everything the app needs, the provision has it, but still no way. 

Installing via XCode keeps returning the same issue stated above:
> ```A signed resource has been added, modified, or deleted```

Also, just downloaded a fresh AIR SDK from [here](https://www.adobe.com/devnet/air/air-sdk-download.html) and tried packaging with that one, nothing.

I have asked for a newer Apple device to try on Monday, I doubt it has to anything to do with that but I have to give it a go.

I'm probably going to leave this aside for a while because it's taking way too long and honestly don't know what else to try.
